### PR TITLE
Address database creation warnings

### PR DESCRIPTION
**Pull request by michplunkett**
## Description of Changes
Addressed DB warnings that we'd see during creation and removed redundant/unnecessary Model properties.
```console
SAWarning: relationship 'Face.users' will copy column users.id to column faces.user_id, which conflicts with relationship(s): 'Face.user' (copies users.id to faces.user_id), 'User.faces' (copies users.id to faces.user_id). If this is not the intention, consider if these relationships should be linked with back_populates, or if viewonly=True should be applied to one or more if they are read-only. For the less common case that foreign key constraints are partially overlapping, the orm.foreign() annotation can be used to isolate the columns that should be written towards.   To silence this warning, add the parameter 'overlaps="faces,user"' to the 'Face.users' relationship.
SAWarning: relationship 'Officer.assignments_lazy' will copy column officers.id to column assignments.officer_id, which conflicts with relationship(s): 'Assignment.officer' (copies officers.id to assignments.officer_id), 'Officer.assignments' (copies officers.id to assignments.officer_id). If this is not the intention, consider if these relationships should be linked with back_populates, or if viewonly=True should be applied to one or more if they are read-only. For the less common case that foreign key constraints are partially overlapping, the orm.foreign() annotation can be used to isolate the columns that should be written towards.   To silence this warning, add the parameter 'overlaps="assignments,officer"' to the 'Officer.assignments_lazy' relationship. (Background on this error at: https://sqlalche.me/e/14/qzyx)
  department = Department(
SAWarning: relationship 'Assignment.base_officer' will copy column officers.id to column assignments.officer_id, which conflicts with relationship(s): 'Assignment.officer' (copies officers.id to assignments.officer_id), 'Officer.assignments' (copies officers.id to assignments.officer_id), 'Officer.assignments_lazy' (copies officers.id to assignments.officer_id). If this is not the intention, consider if these relationships should be linked with back_populates, or if viewonly=True should be applied to one or more if they are read-only. For the less common case that foreign key constraints are partially overlapping, the orm.foreign() annotation can be used to isolate the columns that should be written towards.   To silence this warning, add the parameter 'overlaps="assignments,assignments_lazy,officer"' to the 'Assignment.base_officer' relationship. (Background on this error at: https://sqlalche.me/e/14/qzyx)
  department = Department(
SAWarning: relationship 'Image.users' will copy column users.id to column raw_images.created_by, which conflicts with relationship(s): 'Image.user' (copies users.id to raw_images.created_by), 'User.raw_images' (copies users.id to raw_images.created_by). If this is not the intention, consider if these relationships should be linked with back_populates, or if viewonly=True should be applied to one or more if they are read-only. For the less common case that foreign key constraints are partially overlapping, the orm.foreign() annotation can be used to isolate the columns that should be written towards.   To silence this warning, add the parameter 'overlaps="raw_images,user"' to the 'Image.users' relationship. (Background on this error at: https://sqlalche.me/e/14/qzyx)
  department = Department(
SAWarning: relationship 'User.classifications' will copy column users.id to column raw_images.created_by, which conflicts with relationship(s): 'Image.user' (copies users.id to raw_images.created_by), 'User.raw_images' (copies users.id to raw_images.created_by). If this is not the intention, consider if these relationships should be linked with back_populates, or if viewonly=True should be applied to one or more if they are read-only. For the less common case that foreign key constraints are partially overlapping, the orm.foreign() annotation can be used to isolate the columns that should be written towards.   To silence this warning, add the parameter 'overlaps="raw_images,user"' to the 'User.classifications' relationship. (Background on this error at: https://sqlalche.me/e/14/qzyx)
  department = Department(
SAWarning: relationship 'Face.users' will copy column users.id to column faces.created_by, which conflicts with relationship(s): 'Face.user' (copies users.id to faces.created_by), 'User.faces' (copies users.id to faces.created_by). If this is not the intention, consider if these relationships should be linked with back_populates, or if viewonly=True should be applied to one or more if they are read-only. For the less common case that foreign key constraints are partially overlapping, the orm.foreign() annotation can be used to isolate the columns that should be written towards.   To silence this warning, add the parameter 'overlaps="faces,user"' to the 'Face.users' relationship. (Background on this error at: https://sqlalche.me/e/14/qzyx)
  department = Department(
SAWarning: relationship 'User.tags' will copy column users.id to column faces.created_by, which conflicts with relationship(s): 'Face.user' (copies users.id to faces.created_by), 'User.faces' (copies users.id to faces.created_by). If this is not the intention, consider if these relationships should be linked with back_populates, or if viewonly=True should be applied to one or more if they are read-only. For the less common case that foreign key constraints are partially overlapping, the orm.foreign() annotation can be used to isolate the columns that should be written towards.   To silence this warning, add the parameter 'overlaps="faces,user"' to the 'User.tags' relationship. (Background on this error at: https://sqlalche.me/e/14/qzyx)
```

Relevant information: https://stackoverflow.com/a/59920780

## Tests and linting
 - [x] This branch is up-to-date with the `develop` branch.
 - [x] `pytest` passes on my local development environment.
 - [x] `pre-commit` passes on my local development environment.

The warnings are gone:
```console
Postgres is up
## Populate database with test data
docker-compose run --rm web python ./test_data.py -p
WARN[0000] The "APPROVE_REGISTRATIONS" variable is not set. Defaulting to a blank string. 
[+] Creating 1/0
 ✔ Container openoversight-postgres-1  Running                                                                                                                                                             0.0s 
/usr/local/lib/python3.11/site-packages/flask_limiter/extension.py:293: UserWarning: Using the in-memory storage for tracking rate limits as no storage was explicitly specified. This is not recommended for production use. See: https://flask-limiter.readthedocs.io#configuring-a-storage-backend for documentation about configuring the storage backend.
  warnings.warn(
[2023-08-08 18:05:22,006] INFO in __init__: OpenOversight startup
/usr/local/lib/python3.11/site-packages/flask_limiter/extension.py:293: UserWarning: Using the in-memory storage for tracking rate limits as no storage was explicitly specified. This is not recommended for production use. See: https://flask-limiter.readthedocs.io#configuring-a-storage-backend for documentation about configuring the storage backend.
  warnings.warn(
[2023-08-08 18:05:22,097] INFO in __init__: OpenOversight startup
[*] Populating database with test data...
2023-08-08 18:05:27,443 INFO sqlalchemy.engine.Engine select pg_catalog.version()
```

### CODE REVIEW

**Review by null-space-0 — CHANGES_REQUESTED** [review]
Looks good as well!!

File: OpenOversight/app/models/database.py [hunk 1]
```diff
@@ -334,7 +327,7 @@ class Assignment(BaseModel):
 
     id = db.Column(db.Integer, primary_key=True)
     officer_id = db.Column(db.Integer, db.ForeignKey("officers.id", ondelete="CASCADE"))
-    base_officer = db.relationship("Officer")
+    base_officer = db.relationship("Officer", back_populates="assignments")
```
Comment (b-meson) [inline OpenOversight/app/models/database.py:330]: TIL about `back_populates`

**Review by null-space-0 — APPROVED** [review]
Looks good to me *fixed

**Review by b-meson — APPROVED** [review]

File: OpenOversight/app/models/database.py (hunk 1 above)
Comment (michplunkett) [reply OpenOversight/app/models/database.py:330]: Ayyyyye, that was yesterday for me!

**Pull request merged**